lib/components/ClinicUserEdit.js

```diff
@@ -0,0 +1,234 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2014, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+var React = require('react');
+var _ = require('lodash');
+var sundial = require('sundial');
+
+var config = require('../config');
+
+var styles = require('../../styles/components/ClinicUserEdit.module.less');
+
+var MONTHS = [
+  {value: '', label: 'Month'},
+  {value: '01', label: 'January'},
+  {value: '02', label: 'February'},
+  {value: '03', label: 'March'},
+  {value: '04', label: 'April'},
+  {value: '05', label: 'May'},
+  {value: '06', label: 'June'},
+  {value: '07', label: 'July'},
+  {value: '08', label: 'August'},
+  {value: '09', label: 'September'},
+  {value: '10', label: 'October'},
+  {value: '11', label: 'November'},
+  {value: '12', label: 'December'}
+];
+
+var ClinicUserEdit = React.createClass({
+  propTypes: {
+    errorMessage: React.PropTypes.string,
+    allUsers: React.PropTypes.object.isRequired,
+    loggedInUser: React.PropTypes.string.isRequired,
+    targetId: React.PropTypes.string,
+    updateUser: React.PropTypes.func.isRequired,
+    createUser: React.PropTypes.func.isRequired,
+    cancelEdit: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function(){
+    if(this.props.targetId){
+      var user = _.get(this.props.allUsers, this.props.targetId);
+      return {
+        name: _.get(user, 'fullName'),
+        year: _.get(user, ['patient', 'birthday'], '').substr(0,4),
+        month: _.get(user, ['patient', 'birthday'], '').substr(5,2),
+        day: _.get(user, ['patient', 'birthday'], '').substr(8,2),
+        email: _.get(user, ['patient', 'email'], ''),
+        mrn: _.get(user, ['patient', 'mrn'], '')
+      };
+    }
+    return {
+      name: '',
+      year: '',
+      month: '',
+      day: ''
+    };
+  },
+
+  zeroPad: function(value){
+    return (value.length === 1 ? '0': '') + value;
+  },
+
+  isDisabled: function() {
```

Comment:
It is dense. It started out simple but kept growing and I didn't break it out when I probably should have. It was a conscious decision, mostly because it's going to run this on every form onChange event (every keystroke, essentially) and I personally hate it when forms tell me my input is invalid when I'm in the middle of actually typing it it.